pkgs/build-support/rust/import-cargo-lock.nix

```diff
@@ -79,17 +81,16 @@ let
   # We can't use the existing fetchCrate function, since it uses a
   # recursive hash of the unpacked crate.
   fetchCrate = pkg:
-    assert lib.assertMsg (pkg ? checksum) ''
+    let
+      checksum = pkg.checksum or parsedLockFile.metadata."checksum ${pkg.name} ${pkg.version} (${pkg.source})";
```

Comment:
we might be able to look at the version key, but this will also do

Reply:
Fun fact: there *is* no version key, until v3 (see [here](https://github.com/rust-lang/cargo/blob/a49049817d84f1bfd278de4efe52fa4e4fb43e03/src/cargo/core/resolver/resolve.rs#L49) for an explanation), so this is the simplest way (as far as I know, anyways) that handles all current versions of the format.

That begs the question as to if I should add some sort of comment explaining this -- thoughts?

Reply:
> Fun fact: there is no version key until v3

TIL `version = 2` isn't a thing, this is probably the only way to implement this then.

> That begs the question as to if I should add some sort of comment explaining this 

It would be helpful to comment that `pkg.checksum` is for v2/v3 and `metadata` is for v1, It's up to you if you want to add the comments